deeplake/core/dataset/deeplake_query_tensor.py

```diff
@@ -34,6 +34,8 @@ def __init__(
         self._index = index or Index(self.indra_tensor.index)
 
     def __getattr__(self, key):
+        if key == "chunk_engine":
```

Comment:
Hey @FayazRahman this change is related to the logic in the following part - https://github.com/activeloopai/deeplake/blob/main/deeplake/core/tensor.py#L999 which checks `hasattr`. Would you suggest to change that code something like this?
```
try:
    self.chunk_engine.numpy ...
catch NotImplementedError:
    self.numpy ...
```

Reply:
With the current code, the `hasattr` check will always pass actually, since `Tensor` is base class. Best thing to do would be to have a better condition instead of `hasattr(self, "chunk_engine")`, maybe some `isinstance` check? Otherwise the `try...except` works fine as well.